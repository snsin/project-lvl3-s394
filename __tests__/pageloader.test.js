import nock from 'nock';
import { promises as fs } from 'fs';
import { join } from 'path';
import os from 'os';
import loadPage from '../src';

const fixtures = {};

beforeAll(async () => {
  nock.disableNetConnect();
  fixtures.filename = 'hexlet-io-courses.html';
  fixtures.url = new URL('https://hexlet.io/courses');
  const fixtureDir = '__tests__/__fixtures__/';
  fixtures.data = await fs.readFile(join(fixtureDir, fixtures.filename), 'utf8');
});

beforeEach(async () => {
  fixtures.targetDir = await fs.mkdtemp(join(os.tmpdir(), 'page-load-'));
});


test('load data to file', async () => {
  nock(fixtures.url.origin)
    .get(fixtures.url.pathname)
    .reply(200, fixtures.data);
  await loadPage(fixtures.url.href, fixtures.targetDir);
  const loadedData = await fs.readFile(join(fixtures.targetDir, fixtures.filename), 'utf8');
  expect(loadedData).toBe(fixtures.data);
});

test('page not found', async () => {
  nock(fixtures.url.origin)
    .get(fixtures.url.pathname)
    .reply(404);
  await expect(loadPage(fixtures.url.href, fixtures.targetDir))
    .rejects.toThrow('404');
});

test('directory not exist', async () => {
  nock(fixtures.url.origin)
    .get(fixtures.url.pathname)
    .reply(200, fixtures.data);
  await expect(loadPage(fixtures.url.href, join(fixtures.targetDir, 'not-exist')))
    .rejects.toThrow('ENOENT');
});
