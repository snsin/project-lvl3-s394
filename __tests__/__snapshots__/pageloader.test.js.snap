// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`no connection error 1`] = `"Network error: Nock: Disallowed net connect for \\"jestjs.io:443/en/help\\" https://jestjs.io/en/help"`;

exports[`page not found 1`] = `"Network error: Request failed with status code 404 https://jestjs.io/en/help"`;
